File: Database_PHP/api/Medicine.php
<?php
    class Medicine{

        public function __construct($con)
        {
            $this->con=$con;
        }

        public function insertMedicine($name,$typeInsulin, $amount, $userID, $measureTime,$unit, $note){

            $result=mysqli_query($this->con, "INSERT INTO medicine VALUES ('','$name','$typeInsulin','$amount','$userID','$measureTime','$unit','$note')");
            return $result;
        }

        public function updateMedicine($id,$name,$typeInsulin, $amount, $measureTime,$unit, $note){
            $query = "UPDATE medicine SET name='$name', typeInsulin='$typeInsulin', amount='$amount', measureTime='$measureTime', unit='$unit', note='$note'  WHERE id='$id'";
            $result=mysqli_query($this->con, $query);
            return $result;
        }
    }
?>

File: lib/models/medicine.dart

class MedicineModel{
  final String id;
  final String name;
  final String typeInsulin;
  final String amount;
  final String userID;
  final DateTime measureTime;
  final String unit;
  final String note;
  final String idModel;

  MedicineModel({this.id, this.name, this.typeInsulin, this.amount,this.userID,this.measureTime,
    this.unit, this.note, this.idModel});

  factory MedicineModel.fromJson(Map<String, dynamic> json) {
    return MedicineModel(
      id: json['id'],
      name: json['name'].toString(),
      typeInsulin: json['typeInsulin'].toString(),
      amount: json['amount'].toString(),
      userID: json['userID'].toString(),
      measureTime: DateTime.parse(json['measureTime'].toString()),
      unit: json['unit'].toString(),
      note: json['note'].toString(),
      idModel: "2",
    );
  }
}